plan/predicate_push_down.go

```diff
@@ -133,8 +133,14 @@ func (p *LogicalJoin) PredicatePushDown(predicates []expression.Expression) (ret
 	case InnerJoin:
 		p.LeftConditions = nil
 		p.RightConditions = nil
-		p.EqualConditions = equalCond
-		p.OtherConditions = otherCond
+		p.EqualConditions = make([]*expression.ScalarFunction, 0, len(equalCond))
```

Comment:
`p.EqualConditions = make([]*expression.ScalarFunction, len(equalCond))` ? 

We need fulfill expression later anyway.

Reply:
Will this change affect too much?

Reply:
I think it's ok to use `append`.
We don't need to think if make full length is applicable, we can safely use append.

Reply:
I won't change as much. It is just neat to use the way I proposed.

Reply:
And yes. It's perfect fine to use append.